### PR TITLE
Adding Kubectl Support for Endpoint Slices

### PR DESCRIPTION
**Pull request by robscott**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This adds kubectl support for Endpoint Slices for the new Discovery API Group, building on the earlier PR here: https://github.com/kubernetes/kubernetes/pull/80766. 

**Special notes for your reviewer**:
Until the PR this is built on gets merged in, the diff specific to this PR can be found here: https://github.com/robscott/kubernetes/compare/discovery-api...robscott:endpointslice-kubectl

**Does this PR introduce a user-facing change?**:
```release-note
Adds Endpoint Slice support for kubectl when discovery API group is enabled.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
- [KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-network/20190603-EndpointSlice-API.md)
- [Enhancement Issue](https://github.com/kubernetes/enhancements/issues/752)

### CODE REVIEW

**Comment by robscott** [comment]
The get output includes `NAME`, `ADDRESS TYPE`, `PORTS`, `ENDPOINTS`, and `AGE` columns. Similar to Endpoints, we only show 3 ports. Here's some sample output as tested:

Simple example:
```
abcslice.123   IP    80    10.1.2.3,2001:db8::1234:5678   0s
```
Expanded ports and addresses:
```
multiportslice.123   IP    80,443,3000 + 1 more...   10.1.2.3,2001:db8::1234:5678,10.2.3.4 + 1 more...   5m
```

And the describe output looks like this:
```
Name:         foo.123
Namespace:    bar
Labels:       <none>
Annotations:  <none>
AddressType:  IP
Ports:
  Name     Port  Protocol
  ----     ----  --------
  <unset>  80    TCP
Endpoints:
  - Addresses:  1.2.3.4,1.2.3.5
    Conditions:
      Ready:    true
    Hostname:   <unset>
    TargetRef:  Pod/test-123
    Topology:   topology.kubernetes.io/region=us-central1
                topology.kubernetes.io/zone=us-central1-a
  - Addresses:  1.2.3.6,1.2.3.7
    Conditions:
      Ready:    true
    Hostname:   <unset>
    TargetRef:  Pod/test-124
    Topology:   topology.kubernetes.io/region=us-central1
                topology.kubernetes.io/zone=us-central1-b
Events:         <none>
```

File: staging/src/k8s.io/kubectl/pkg/describe/versioned/describe.go [hunk 1]
```diff
@@ -2601,6 +2603,103 @@ func describeEndpoints(ep *corev1.Endpoints, events *corev1.EventList) (string,
...
+	c := d.DiscoveryV1alpha1().EndpointSlices(namespace)
+
+	eps, err := c.Get(name, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	var events *corev1.EventList
+	if describerSettings.ShowEvents {
+		events, _ = d.CoreV1().Events(namespace).Search(scheme.Scheme, eps)
```
Comment (tedyu) [inline staging/src/k8s.io/kubectl/pkg/describe/versioned/describe.go]: Shouldn't the error return be checked ?

**Pull request merged**